src/libtls/tls_server.c

```diff
@@ -1216,7 +1252,7 @@ static status_t send_server_key_exchange(private_tls_server_t *this,
 
 	if (diffie_hellman_group_is_ec(group))
 	{
-		curve = ec_group_to_curve(this, group);
+		curve = tls_ec_group_to_curve(group);
```

Comment:
You should probably move this change to the first commit in this branch (also the removal of the `ec_group_to_curve()` helper).

Using the new helper is not exactly the same, though. The previous code indirectly made sure that the DH group is actually supported. But that was probably redundant, because the group we get here initially will be the DH group from the selected cipher suite and suites with unsupported DH groups should have been filtered out earlier anyway.